cmd/liqonet/main.go

```diff
@@ -236,18 +230,25 @@ func main() {
 			},
 			RetryTimeout: 30 * time.Second,
 		}
-		go r.Watcher(r.DynClient, peeringReqGVR, r.PeeringRequestHandler, r.PReqWatcher)
-		go r.Watcher(r.DynClient, advGVR, r.AdvertisementHandler, r.AdvWatcher)
+		//starting the watchers
+		go r.Watcher(r.DynFactory, liqonetOperators.AdvGVR, cache.ResourceEventHandlerFuncs{
+			AddFunc:    r.AdvertisementHandlerAdd,
+			UpdateFunc: r.AdvertisementHandlerUpdate,
+		}, r.AdvStartWatcher, r.AdvStopWatcher)
+		go r.Watcher(r.DynFactory, liqonetOperators.PeeringReqGVR, cache.ResourceEventHandlerFuncs{
+			AddFunc:    r.PeeringRequestHandlerAdd,
+			UpdateFunc: r.PeeringRequestHandlerUpdate,
```

Comment:
Don't we need to react also to delete events?

Reply:
this is a little bit complicated. Just because a peeringrequests or and advertisement is deleted  we should not tear down the network connection between the two clusters. Im thinking how to integrate the network connection in the "unjoin" flow.